clcache.py

```diff
@@ -232,9 +232,21 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         # One of the few exceptions to this rule is the /MP switch, which only
         # defines how many compiler processes are running simultaneusly.
         commandLine = [arg for arg in commandLine if not arg.startswith("/MP")]
+        arguments, inputFiles = CommandLineAnalyzer.parseArgumentsAndInputFiles(commandLine)
+        collapseBasedirInCmdPath = lambda path: collapseBasedirToPlaceholder(os.path.normcase(os.path.abspath(path)))
+
+        commandLineArgs = []
+        projectSpecificArgs = ("AI", "I", "FU")
+        for k in sorted(arguments.keys()):
```

Comment:
Actually the sorted is very important to have repeatable manifest hashes. As dictionaries in python are unordered the only way to guarantee that we generate the same hash every invocation is to iterate over the keys in the same order. It is the same case that #235 solved.